src/lib/message/L2ToL1Message.ts

```diff
@@ -66,35 +65,36 @@ export enum L2ToL1MessageStatus {
 export type L2ToL1MessageReaderOrWriter<T extends SignerOrProvider> =
   T extends Provider ? L2ToL1MessageReader : L2ToL1MessageWriter
 
+export type L2ToL1TransactionEvent = {
+  caller: string
+  destination: string
+  hash: BigNumber
+  position: BigNumber
+  indexInBatch?: BigNumber
+  arbBlockNum: BigNumber
+  ethBlockNum: BigNumber
+  timestamp: BigNumber
+  callvalue: BigNumber
+  data: string
+}
+
 /**
  * Base functionality for L2->L1 messages
  */
 export class L2ToL1Message {
   public static fromEvent<T extends SignerOrProvider>(
     l1SignerOrProvider: T,
-    event: L2ToL1TransactionEvent['args'],
-    outboxAddress: string,
-    isNitro: boolean
+    event: L2ToL1TransactionEvent,
```

Comment:
we can accept here either the classic or nitro event, I think its cleaner to use the union of those types here
```suggestion
    event: ClassicL2ToL1TransactionEvent['args'] | NitroL2ToL1TransactionEvent['args'],
```